crates/bevy_reflect/src/array.rs

```diff
@@ -0,0 +1,207 @@
+use crate::{serde::Serializable, Reflect, ReflectMut, ReflectRef};
+use std::{
+    any::Any,
+    hash::{Hash, Hasher},
+};
+
+/// A static-sized array of [`Reflect`] items.
+/// This corresponds to types like `[T; N]` (arrays)
+pub trait Array: Reflect {
+    fn get(&self, index: usize) -> Option<&dyn Reflect>;
+    fn get_mut(&mut self, index: usize) -> Option<&mut dyn Reflect>;
+    fn len(&self) -> usize;
+    fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+    fn iter(&self) -> ArrayIter;
+    fn clone_dynamic_array(&self) -> DynamicArray {
+        DynamicArray {
+            name: self.type_name().to_string(),
+            values: self.iter().map(|value| value.clone_value()).collect(),
+        }
+    }
+}
+
+pub struct DynamicArray {
+    pub(crate) name: String,
+    pub(crate) values: Box<[Box<dyn Reflect>]>,
+}
+
+impl DynamicArray {
+    #[inline]
+    pub fn new(values: Box<[Box<dyn Reflect>]>) -> Self {
+        Self {
+            name: String::default(),
+            values,
+        }
+    }
+
+    #[inline]
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    #[inline]
+    pub fn set_name(&mut self, name: String) {
+        self.name = name;
+    }
+}
+
+// SAFE: any and any_mut both return self
+unsafe impl Reflect for DynamicArray {
+    #[inline]
+    fn type_name(&self) -> &str {
+        self.name.as_str()
+    }
+
+    #[inline]
+    fn any(&self) -> &dyn Any {
+        self
+    }
+
+    #[inline]
+    fn any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+
+    fn apply(&mut self, value: &dyn Reflect) {
+        array_apply(self, value);
+    }
+
+    #[inline]
+    fn set(&mut self, value: Box<dyn Reflect>) -> Result<(), Box<dyn Reflect>> {
+        *self = value.take()?;
+        Ok(())
+    }
+
+    #[inline]
+    fn reflect_ref(&self) -> ReflectRef {
+        ReflectRef::Array(self)
+    }
+
+    #[inline]
+    fn reflect_mut(&mut self) -> ReflectMut {
+        ReflectMut::Array(self)
+    }
+
+    #[inline]
+    fn clone_value(&self) -> Box<dyn Reflect> {
+        Box::new(self.clone_dynamic_array())
+    }
+
+    #[inline]
+    fn reflect_hash(&self) -> Option<u64> {
+        array_hash(self)
+    }
+
+    fn reflect_partial_eq(&self, value: &dyn Reflect) -> Option<bool> {
+        array_partial_eq(self, value)
+    }
+
+    fn serializable(&self) -> Option<Serializable> {
+        None
```

Comment:
If the inner elements are serializable, shouldn't the array be serializable too?

Reply:
This was just an artifact of copying the `List` implementation. 
however, yes, they should be :)